src/code42cli/cmds/devices.py

```diff
@@ -60,6 +64,20 @@ def change_device_name_option(help_msg):
 )
 
 
+@devices.command()
+@device_guid_argument
+@new_device_name_option
+@sdk_options()
+def rename(state, device_guid, new_device_name):
```

Comment:
Let me know if we want to add any other checks here (ie. should users not be able to rename a device in legal hold?)